apis/python/src/tiledbsoma/options/soma_tiledb_context.py

```diff
@@ -96,3 +97,16 @@ def __attrs_post_init__(self) -> None:
         else:
             object.__setattr__(self, "_group_write_tiledb_ctx", self.tiledb_ctx)
         assert isinstance(self._group_write_tiledb_ctx, tiledb.Ctx)
+
+    @classmethod
+    def evolve(cls, context: Self, **changes: Dict[str, Any]) -> Self:
```

Comment:
personally I prefer `merge` but this is fine

Reply:
I am fine with any name choice. `merge` is also good, and `replace` is another frequently-used one (cf. `inspect.Parameter`).

Two levels of suggestion here:

FIRST: A minor change that doesn't change your intent but saves lots of typing:

```python
@attrs.define
class SomeAttrsClass:
  ...

  evolve = attrs.evolve

  # This ends up working equivalently to:
  def evolve(self, **kwargs):
    return attrs.evolve(self, **kwargs)
  # ... because `self` gets bound to the first param of `attrs.evolve`
  # when accessed as `some_instance.evolve(...)`  
```

Given that this solves an immediate need I would be fine with it going in like this (or even the slightly more verbose `def evolve(...)` version).

POTENTIALLY DEEPER SUGGESTION:

That said I think in this case something like this might be a bit more useful (and completely fine if you save this for a follow-up):

```python
@attrs.define(...)
class SOMATileDBContext:

  def evolve(self, *, tiledb_config: Optional[Dict[str, Any]] = None, **changes: Any) -> Self:
    if tiledb_config:
      # build a new tiledb.Ctx here
      changes["tiledb_ctx"] = new_ctx
    attrs.evolve(self, **changes)
```

That would let you say something like `some_inst.evolve(tiledb_config={"put": "your", "overrides": "here"})`

ALSO A MINOR NOTE: When annotating things like `*args` and `**kwargs`, the annotation should reflect the type of *one* of those things, so:

```python
def any_number_of_strings(*whatever: str) -> result:
  ...

any_number_of_strings("this", "that", "the other")

def some_ints(**kwargs: int) -> result:
  ...

some_ints(a=1, b=2, c=3)
```

Reply:
oohhh `replace` even better :D

Reply:
All super good suggestions.  Will implement all, evolving to replace :-)